### PR TITLE
Performant nginx configuration

### PR DESCRIPTION
**Pull request by pierreozoux**
https://developers.google.com/speed/pagespeed/insights/?url=http%3A%2F%2Fblog.indie.host%2F

From https://github.com/indiehosters/known/blob/master/nginx.conf

### CODE REVIEW

File: nginx.conf [hunk 1]
```diff
@@ -1,27 +1,87 @@
...
+    open_file_cache_valid    60s;
+    open_file_cache_min_uses 5;
+    open_file_cache_errors   off;
+    
+    location ~* \.(css|js|jpg|jpeg|gif|png|ico|xml)$ {
+      expires           30d;
+      add_header Pragma public;
+      add_header Cache-Control "public";
+      rewrite ^/css/[0-9]+/(.*) /css/$1 last;
+      rewrite ^/js/[0-9]+/(.*) /js/$1 last;
```
Comment (kylewm) [inline nginx.conf]: I don't think we need these rewrites anymore

File: nginx.conf [hunk 2]
```diff
@@ -1,27 +1,87 @@
...
+      try_files $uri $uri/ /index.php?$query_string;
     }
 
     location ~ \.php$ {
-        try_files $uri =404;
-        include /etc/nginx/fastcgi_params;
-        fastcgi_pass unix:/var/run/php5-fpm.sock;
-        fastcgi_index index.php;
-        fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
+      fastcgi_param  GATEWAY_INTERFACE  CGI/1.1;
```
Comment (kylewm) [inline nginx.conf]: @paulcmal suggested `try_files $uri =404;` here: https://github.com/idno/Known/issues/1031#issuecomment-135958676

Worth keeping it?

File: nginx.conf (hunk 1 above)
Comment (pierreozoux) [inline nginx.conf]: in 0.9.2 I do need them. Do you mean now, in the current version we don't need them anymore?

File: nginx.conf (hunk 2 above)
Comment (pierreozoux) [inline nginx.conf]: You mean for the line 47 and 51?

I just tried, but it is giving me a 404...

Comment (kylewm) [inline nginx.conf]: In `location ~ \.php$ { ... }` line 55, similar to how it was in the old file.

File: nginx.conf (hunk 1 above)
Comment (kylewm) [inline nginx.conf]: Ah ok, yeah the code in master doesn't need them, but I guess there hasn't been a release since then.

File: nginx.conf (hunk 2 above)
Comment (pierreozoux) [inline nginx.conf]: ah ok, just updated it.

File: nginx.conf (hunk 1 above)
Comment (pierreozoux) [inline nginx.conf]: ok, removed.

**Comment by rikmendes** [comment]
Wondering if any of you could also produce a working nginx conf for Known when installed into a sub directory like yourdomain.net/blog/ up to now all configurations I have seen so far are for root installations. 

ps: I tried all the tweaks I could find last weekend to get it running on a subdir without any success.

**Comment by pierreozoux** [comment]
\o/

**Pull request merged**